socorro/scripts/reprocess.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import argparse
+import os
+import sys
+import time
+import urlparse
+
+import requests
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+from socorro.scripts import WrappedTextHelpFormatter
+
+
+DESCRIPTION = """
+Sends specified crashes for reprocessing
+
+This requires SOCORRO_REPROCESS_API_TOKEN to be set in the environment to a valid API token.
+
+"""
+
+DEFAULT_HOST = 'https://crash-stats.mozilla.com'
+CHUNK_SIZE = 50
+SLEEP = 1
+
+
+def session_with_retries(url):
+    base_url = urlparse.urlparse(url).netloc
+    scheme = urlparse.urlparse(url).scheme
+
+    retries = Retry(total=32, backoff_factor=1, status_forcelist=[429])
+
+    s = requests.Session()
+    s.mount(scheme + '://' + base_url, HTTPAdapter(max_retries=retries))
+
+    return s
+
+
+def main(argv):
+    parser = argparse.ArgumentParser(
+        formatter_class=WrappedTextHelpFormatter,
+        prog=os.path.basename(__file__),
+        description=DESCRIPTION.strip(),
+    )
+    parser.add_argument('--host', help='host for system to reprocess in', default=DEFAULT_HOST)
+    parser.add_argument('crashid', nargs='*', help='one or more crash ids to fetch data for')
+
+    args = parser.parse_args(argv)
+
+    # FIXME(willkg): Rethink this?
```

Comment:
Is leaving this in intentional?

Reply:
I left this in intentionally. I do this a lot. There are a bunch of things I want to think about, but I've been pushing to get things done and deferring thinking about them especially with things where we don't need to figure it out now. In some cases, it helps to wait to see how other things nearby gel.

The issue here is that now we have two API tokens with different permissions. Is it better to have a local development environment and require the API token to have *all* the permissions, or better to have a bunch of different environment variables for the various tokens we need.

Having said all that, this particular line isn't wildly helpful for anyone who isn't me. I'll remove it.